backend/server.js

```diff
@@ -61,17 +62,62 @@ hapi.ext({
 sbp('okTurtles.data/set', SERVER_INSTANCE, hapi)
 
 sbp('sbp/selectors/register', {
-  'backend/server/broadcastEntry': async function (entry: GIMessage) {
+  'backend/server/persistState': async function (deserializedHEAD: Object, entry: string) {
+    const contractID = deserializedHEAD.contractID
+    const cheloniaState = sbp('chelonia/private/state')
+    // If the contract has been removed or the height hasn't been updated,
+    // there's nothing to persist
+    if (!cheloniaState.contracts[contractID] || cheloniaState.contracts[contractID].height < deserializedHEAD.head.height) {
+      return
+    }
+    // If the current HEAD is not what we expect, don't save (the state could
+    // have been updated by a later message). This ensures that we save the
+    // latest state and also reduces the number of write operations
+    if (cheloniaState.contracts[contractID].HEAD === deserializedHEAD.hash) {
+      // Extract the parts of the state relevant to this contract
+      const state = {
+        contractState: cheloniaState[contractID],
+        cheloniaContractInfo: cheloniaState.contracts[contractID]
+      }
+      // Save the state under a 'contract partition' key, so that updating a
+      // contract doesn't require saving the entire state.
+      // Although it's not important for the server right now, this will fail to
+      // persist changes the state for other contracts (for example, this happens
+      // when watching keys)
+      await sbp('chelonia/db/set', '_private_cheloniaState_cp_' + contractID, JSON.stringify(state))
```

Comment:
One more question: the request from before was to use `head=` to store this info, to avoid having to do a few extra writes to the database per contract even update.

This PR doesn't seem to do that, what is the reason why?

Note that `head=` is still being used in addition to the code here, and so the latestHEADInfo is being stored twice...